src/core/public/ui_settings/ui_settings_api.ts

```diff
@@ -55,10 +56,9 @@ const NOOP_CHANGES = {
 export class UiSettingsApi {
   private pendingChanges?: Changes;
   private sendInProgress = false;
-
   private readonly loadingCount$ = new BehaviorSubject(0);
 
-  constructor(private readonly http: HttpSetup) {}
+  constructor(private readonly http: HttpSetup, private readonly scope?: UiSettingScope) {}
```

Comment:
```suggestion
  constructor(private readonly http: HttpSetup, private readonly scope: UiSettingScope = Scope.Global) {}
```
So that we do not add check if this.scope exist in other method.

Reply:
We can't default the scope to global when it's not provided, because user-level settings also omit the scope.
In such cases, we should let the server side to decide which scope it belongs

Reply:
We can't default the scope to global when it's not explicitly provided, because user-level settings also omit the scope.
In such cases, we should delegate the decision to the server, allowing it to determine the appropriate scope based on the context.

Reply:
I see, thanks.